src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -290,8 +300,8 @@ object BoringUtils {
       Builder.error(s"Cannot bore from $source to ${thisModule.name}, as they do not share a least common ancestor")
     }
     val (upPath, downPath) = lcaResult.get
-    val lcaSource = drill(source, upPath.dropRight(1), upPath.dropRight(1), true)
-    val sink = drill(lcaSource, downPath.reverse.tail, downPath.reverse, false)
+    val lcaSource = drill(source, upPath.dropRight(1), upPath.dropRight(1), true && !isDrive)
+    val sink = drill(lcaSource, downPath.reverse.tail, downPath.reverse, false || isDrive)
```

Comment:
Interesting choice of representing this that way. I assume this is done because it is perceived as more understandable than doing the boolean optimization?

```suggestion
    val lcaSource = drill(source, upPath.dropRight(1), upPath.dropRight(1), up=!isDrive)
    val sink = drill(lcaSource, downPath.reverse.tail, downPath.reverse, up=isDrive)
```

Reply:
Interesting is putting it politely... I added without thinking about the redundancy.